### PR TITLE
Add `conflicts: true` option to mango selectors

### LINKED ISSUE #1101: Mango queries with `conflicts: true` return documents that do not have conflicts
Queries to the `/{db}/_find` endpoint that set `conflicts: true` return documents that do not include conflicts. I have documented a test case [here](https://gist.github.com/garbados/335bc7c1836cbf726a6b511d6c0fa418) that reproduces the issue.

The documentation in [cloudant/mango]() indicates that this is expected behavior:

> conflicts (optional) (default: false) - boolean, whether or not to include information about any existing conflicts for the document.

However, it sure would be nice to issue queries that return *only* conflicted documents that match the given selector.

## Expected Behavior

A query setting should exist to filter query results to only those that have conflicted and match the given selector.

## Current Behavior

Setting `conflicts: true` returns documents with and without conflicts, such that the user must perform a second query to `/_all_docs` to determine which documents actually have conflicts.

## Possible Solution

Setting `onlyConflicts` or otherwise-named query value should cause CouchDB to filter results of a query to only those documents that match a given selector *and* which have conflicts.

## Steps to Reproduce (for bugs)

I've developed a test case [here](https://gist.github.com/garbados/335bc7c1836cbf726a6b511d6c0fa418) using node.js.

## Context

Detecting conflicts requires finding them first. In large databases, performing one query to find documents that may have conflicts and then another to determine which of them actually do creates considerable request overhead for applications attempting to detect or resolve conflicts proactively.

## Your Environment

* Version used: 2.1.1-180a155
* Browser Name and version: N/A
* Operating System and version (desktop or mobile): Ubuntu 16.04.3 LTS
* Link to your project: [gist](https://gist.github.com/garbados/335bc7c1836cbf726a6b511d6c0fa418)

### PR DESCRIPTION
**Pull request by janl**
This allows for using Mango queries for finding docs with conflicts.

Closes #1101

### CODE REVIEW

**Comment by janl** [comment]
script to repro:

```
#!/bin/sh -ex
S=http://a:a@127.0.0.1:15984
DB=x1101

# delete test db
curl -X DELETE $S/$DB

# create test d
curl -X PUT $S/$DB

# create doc

curl -X PUT $S/$DB/doc -d '{"a":2}'

# create conflict
curl -X PUT $S/$DB/doc?new_edits=false -d '{"a":1,"_rev":"1-23202479633c2b380f79507a776743d5"}'

# validate we have created a conflict:
curl $S/$DB/doc?conflicts=true

# create mango index def
curl -X POST $S/$DB/_index -d '{"index":{"fields": ["_conflicts"]}}' -Hcontent-type:application/json

# query mango index
curl -X POST $S/$DB/_find -d '{"selector": {"_conflicts": { "$exists": true}},"conflicts":true}' -Hcontent-type:application/json

# query mango index without conflicts option
curl -X POST $S/$DB/_find -d '{"selector": {"_conflicts": { "$exists": true}}}' -Hcontent-type:application/json
```

Successful run:

```
 ~/t.sh
+ S=http://a:a@127.0.0.1:15984
+ DB=x1101
+ curl -X DELETE http://a:a@127.0.0.1:15984/x1101
{"ok":true}
+ curl -X PUT http://a:a@127.0.0.1:15984/x1101
{"ok":true}
+ curl -X PUT http://a:a@127.0.0.1:15984/x1101/doc -d '{"a":2}'
{"ok":true,"id":"doc","rev":"1-3975759ccff3842adf690a5c10caee42"}
+ curl -X PUT 'http://a:a@127.0.0.1:15984/x1101/doc?new_edits=false' -d '{"a":1,"_rev":"1-23202479633c2b380f79507a776743d5"}'
{"ok":true,"id":"doc","rev":"1-23202479633c2b380f79507a776743d5"}
+ curl 'http://a:a@127.0.0.1:15984/x1101/doc?conflicts=true'
{"_id":"doc","_rev":"1-3975759ccff3842adf690a5c10caee42","a":2,"_conflicts":["1-23202479633c2b380f79507a776743d5"]}
+ curl -X POST http://a:a@127.0.0.1:15984/x1101/_index -d '{"index":{"fields": ["_conflicts"]}}' -Hcontent-type:application/json
{"result":"created","id":"_design/42ac6b715ea8fb5be44a74e6cf515eaa334238d6","name":"42ac6b715ea8fb5be44a74e6cf515eaa334238d6"}
+ curl -X POST http://a:a@127.0.0.1:15984/x1101/_find -d '{"selector": {"_conflicts": { "$exists": true}},"conflicts":true}' -Hcontent-type:application/json
{"docs":[
{"_id":"doc","_rev":"1-3975759ccff3842adf690a5c10caee42","a":2,"_conflicts":["1-23202479633c2b380f79507a776743d5"]}
],
"bookmark": "g1AAAABheJzLYWBgYMpgSmHgKy5JLCrJTq2MT8lPzkzJBYozA1kgKQ6YVA5QkBFMgKSVDHWNjI0MjEzMLc2MjZONkowtDNLMLU0NzBPNzc3MTYxTTLOysgCY2ReV"}

+ curl -X POST http://a:a@127.0.0.1:15984/x1101/_find -d '{"selector": {"_conflicts": { "$exists": true}}}' -Hcontent-type:application/json
{"docs":[
],
"bookmark": "nil"}
```

**Comment by wohali** [comment]
test case in the mango test suite please

**Comment by janl** [comment]
Would love some help here. I have a hard time grocking that particular flavour of python.

**Comment by wohali** [comment]
LGTM but I'd want to see a +1 from garren or tonysun or similar :)

**Comment by wohali** [comment]
Also docs update please :cat:

File: src/mango/test/19-find-conflicts.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+]
+
+class ChooseCorrectIndexForDocs(mango.DbPerClass):
+    def setUp(self):
+        self.db.recreate()
+        self.db.save_docs(copy.deepcopy(DOC))
+        self.db.save_docs_with_conflicts(copy.deepcopy(CONFLICT))
+
+    def test_retrieve_conflicts(self):
+        self.db.create_index(["_conflicts"])
```
Comment (garrensmith) [inline src/mango/test/19-find-conflicts.py]: This test doesn't need this index since it won't actually use it with an `$exists` query.

Comment (wohali) [reply src/mango/test/19-find-conflicts.py]: @garrensmith sorry to ask this so late, but are you saying that all `$exists` queries always perform a full DB scan and ignore secondary indexes?

**Comment by garrensmith** [comment]
No, not all all. Exists: false will affect index selection since no field with exists false can be in an index.


________________________________

**Pull request merged**